File: ts/impots-entreprise-kata/src/api/calculateur.impot.api.ts
import {Controller, Get} from '@nestjs/common';
import {CalculateurImpotService} from '../service/calculateur-impot.service';
import {Query} from "@nestjs/common/utils/decorators/route-params.decorator";
import {ApiImplicitQuery, ApiUseTags} from "@nestjs/swagger";
import {EntrepriseInfoService} from "../service/entreprise.info.service";
import {Entreprise} from "../domain/entreprise";

@Controller()
@ApiUseTags('tax-calculator-api')
export class CalculateurImpotApi {
    constructor(
        private readonly calculateurImpotService: CalculateurImpotService,
        private readonly entrepriseInfoService: EntrepriseInfoService) {
    }

    @Get('/calculImpot')
    @ApiImplicitQuery({name: 'siret', description: 'Numero de siret', required: true})
    @ApiImplicitQuery({name: 'denomination', description: 'Dénomination de l\'entrepise', required: true})
    calculerImpot(@Query('siret') siret: string, @Query('denomination') denomination: string) {
        const entreprise: Entreprise = this.entrepriseInfoService.getEntrepise(siret, denomination);
        const chiffreAffaire: number = this.entrepriseInfoService.getChiffreAffaire(entreprise);

        return this.calculateurImpotService.calculerImpot(entreprise, chiffreAffaire);
    }
}

File: java/impots-entreprise-kata/src/main/java/fr/soat/cleancoders/api/CalculateurImpotAPI.java
package fr.soat.cleancoders.api;

import fr.soat.cleancoders.domain.Entreprise;
import fr.soat.cleancoders.services.EntrepriseInfoService;
import fr.soat.cleancoders.services.CalculateurImpotService;
import io.swagger.annotations.Api;
import io.swagger.annotations.ApiOperation;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.RequestParam;


@Controller
@RequestMapping("/")
@Api(value = "TaxCalculator", description = "Tax computation API")
public class CalculateurImpotAPI {

    private CalculateurImpotService calculatorService;
    private EntrepriseInfoService entrepriseInfoService;

    public CalculateurImpotAPI(CalculateurImpotService calculatorService, EntrepriseInfoService entrepriseInfoService) {
        this.calculatorService = calculatorService;
        this.entrepriseInfoService = entrepriseInfoService;
    }

    @ApiOperation(value = "Tax computation")
    @RequestMapping(value = "/taxes", method = RequestMethod.GET)
    public Double calculerImpot(@RequestParam(value = "siretNumber") String siretNumber,
                                @RequestParam(value = "denomination") String denomination) {

        Entreprise entreprise = entrepriseInfoService.getEntrepise(siretNumber, denomination);
        Double chiffreAffaire = entrepriseInfoService.getChiffreAffaire();

        return calculatorService.calculerImpot(entreprise, chiffreAffaire);
    }
}
